### PR TITLE
Zeppelin spark

### PR DESCRIPTION
**Pull request by bpanneton**
Get YARN working with updated Zeppelin's Spark Interpreter

### CODE REVIEW

File: magpie/setup/magpie-setup-project-zeppelin [hunk 1]
```diff
@@ -121,7 +136,15 @@ for j in `ls $spark_jars_path`
 do
     zeppelin_jars="$spark_jars_path/$j,$zeppelin_jars"
 done
-zeppelin_jars="${zeppelin_jars},${ZEPPELIN_HOME}/zeppelin-server-${ZEPPELIN_VERSION}.jar"
+
+Magpie_vercomp ${ZEPPELIN_VERSION} 0.7.2
```
Comment (chu11) [inline magpie/setup/magpie-setup-project-zeppelin]: Could you add ```# Returns 0 for ==, 1 for $1 > $2, 2 for $1 < $2``` again?  Thanks.

File: magpie/setup/magpie-setup-project-zeppelin [hunk 2]
```diff
@@ -101,6 +101,21 @@ fi
...
+    do
+        zeppelin_jars="$j,${zeppelin_jars}"
+    done
+
+    hadoop_common=$(ls "${HADOOP_HOME}"/share/hadoop/common/*.jar)
+    for j in $hadoop_common
+    do
+        zeppelin_jars="$j,${zeppelin_jars}"
+    done
+fi
```
Comment (chu11) [inline magpie/setup/magpie-setup-project-zeppelin]: is every jar really necessary?  I suspect it's a much smaller set.  LIkely the common client code or something like that.

File: magpie/setup/magpie-setup-project-spark [hunk 3]
```diff
@@ -544,6 +544,21 @@ then
...
+        sparkclasspath="$j:$sparkclasspath"
+    done
+
+    hadoop_common=$(ls "${HADOOP_HOME}"/share/hadoop/common/*.jar)
+    for j in ${hadoop_common}
+    do
+        sparkclasspath="$j:$sparkclasspath"
+    done
+fi
+
```
Comment (chu11) [inline magpie/setup/magpie-setup-project-spark]: is every jar really necessary? I suspect it's a much smaller set. LIkely the common client code or something like that.

File: magpie/setup/magpie-setup-project-hadoop [hunk 4]
```diff
@@ -1006,6 +1006,13 @@ then
         yarnauxmapreduceshuffle="yarn.nodemanager.aux-services.mapreduce_shuffle.class"
     fi
 
+    if [ "${YARN_VMEM_CHECK}" == "false" ]
+    then
+        yarnvmemcheck='false'
+    else
+        yarnvmemcheck='true'
+    fi
+
```
Comment (chu11) [inline magpie/setup/magpie-setup-project-hadoop]: I'm not familiar with this environment variable.  Is this coming from Spark or Zeppelin?  Or is this from your code?  Should it be a new config in Magpie?

**Comment by chu11** [comment]
hey @bpanneton, I have a feeling with this PR that in your environment / test you hit something related to vmem limits and disabling yarn.nodemanager.vmem-check-enabled solves it.

But disabling yarn.nodemanager.vmem-check-enabled  seems somewhat of an independent issue not really specific to zeppelin?

Perhaps disabling virtual mem checks in yarn should be a generic magpie configurable thing for Yarn?  like MAGPIE_HADOOP_YARN_VMEM_CHECK=true/false??  And the physical mem check variant as well.

**Comment by bpanneton** [comment]
You were correct, the Yarn related VMEM was unrelated to Zeppelin. I will submit another PR for it. Also, I realized that we don't have to pass all the jars to Zeppelin for spark/hadoop. It figures it out based on the config files.

**Comment by chu11** [comment]
just so there's no confusion, is there an order to the PRs that needs to be applied.  Or are they independent of each other?

**Comment by chu11** [comment]
nevermind, it's now clear this is first.

**Comment by chu11** [comment]
patch looks good to me, got ahead and rebase/push #251 and I'll look at that next.

**Pull request merged**